packages/core/src/State.ts

```diff
@@ -288,3 +289,28 @@ export function cloneState<TState extends AnyState>(
     state.machine
   ) as TState;
 }
+
+export function getPersistedState<TState extends AnyState>(
+  state: TState
+): PersistedMachineState<TState> {
+  const {
+    configuration,
+    transitions,
+    tags,
+    machine,
+    children,
+    ...jsonValues
+  } = state;
+
+  const childrenJson: any = {};
+
+  for (const key in children) {
+    childrenJson[key] = children[key].getPersistedState?.();
+  }
+
+  return {
+    ...jsonValues,
+    children: childrenJson,
+    persisted: true
```

Comment:
q: what's the purpose of this flag? I couldn't find any code that would use this

Reply:
I think while I was working on this PR, it was hard to distinguish between a normal "state config" and a persisted "state config"; the latter would need to rehydrate the `children`

Reply:
how they are different though? why we'd like to ignore the children when rehydrating from a "state config"?

Reply:
The `children` here is a key-value pairing where the value is the persisted state. In a live state config, the `children` values would be the actual actorrefs

Reply:
> In a live state config, the children values would be the actual actorrefs

That is the distinction then - maybe that would allow us to get rid of `persisted: true`.

Reply:
Ye, I think this should be enough to test those things out. `persisted: true` is a little bit redundant + it would be a little bit silly if we'd fail to rehydrate a persisted state that doesn't have this property (when the rest of it is valid, ofc). We already have a lot of validation here at the type-level since `PersistedState` isn't compatible with `State` (and vice-versa)